### ISSUE TITLE
listing files on windows ftp server

### ISSUE THREAD

**Issue by toonknapen** [aio-libs/aioftp#82]
```
import aioftp
import asyncio

async def foo():
    client = aioftp.Client()
    await client.connect("localhost")
    await client.login('ftp', 'me@mine.org')
    dir = await client.list("/")
    print(dir)

task = asyncio.ensure_future(foo())
loop = asyncio.get_event_loop()
loop.run_until_complete(task)
```

The code above prints an empty directory listing. 
However using ftplib as below, I do get a non-empty correct list of files and directories that are located at the root of my ftp server

```
from ftplib import FTP

ftp = FTP('localhost', 'ftp', 'me@mine.org')
ls = ftp.nlst()
print(ls)
```
Anyone can tell me what I'm doing wrong? Or might there be issues because of the windows ftp server?
I'm using python 3.6.5, aioftp==0.11.1 and asyncio==3.4.3 and running on windows 10

**Comment by pohmelie** [comment]
Enable `aioftp` logging and rerun your test, it looks like `list` parsing is broken. What ftp server do you use?

**Comment by toonknapen** [comment]
That gives me following:

```
2018-10-15 20:01:50,024 - asyncio - DEBUG - 10004:20088 - selector_events.py:65 - Using selector: SelectSelector
2018-10-15 20:01:50,048 - aioftp.client - INFO - 10004:20088 - client.py:169 - 220 Microsoft FTP Service
2018-10-15 20:01:50,048 - aioftp.client - INFO - 10004:20088 - client.py:240 - USER ftp
2018-10-15 20:01:50,048 - aioftp.client - INFO - 10004:20088 - client.py:169 - 331 Anonymous access allowed, send identity (e-mail name) as password.
2018-10-15 20:01:50,049 - aioftp.client - INFO - 10004:20088 - client.py:240 - PASS me@mine.org
2018-10-15 20:01:50,050 - aioftp.client - INFO - 10004:20088 - client.py:169 - 230 User logged in.
2018-10-15 20:01:50,050 - aioftp.client - INFO - 10004:20088 - client.py:240 - TYPE I
2018-10-15 20:01:50,051 - aioftp.client - INFO - 10004:20088 - client.py:169 - 200 Type set to I.
2018-10-15 20:01:50,051 - aioftp.client - INFO - 10004:20088 - client.py:240 - EPSV
2018-10-15 20:01:50,051 - aioftp.client - INFO - 10004:20088 - client.py:169 - 229 Entering Extended Passive Mode (|||56181|)
2018-10-15 20:01:50,056 - aioftp.client - INFO - 10004:20088 - client.py:240 - MLSD /
2018-10-15 20:01:50,056 - aioftp.client - INFO - 10004:20088 - client.py:169 - 500 Command not understood.
2018-10-15 20:01:50,057 - aioftp.client - INFO - 10004:20088 - client.py:240 - TYPE I
2018-10-15 20:01:50,057 - aioftp.client - INFO - 10004:20088 - client.py:169 - 200 Type set to I.
2018-10-15 20:01:50,057 - aioftp.client - INFO - 10004:20088 - client.py:240 - EPSV
2018-10-15 20:01:50,058 - aioftp.client - INFO - 10004:20088 - client.py:169 - 229 Entering Extended Passive Mode (|||56183|)
2018-10-15 20:01:50,061 - aioftp.client - INFO - 10004:20088 - client.py:240 - LIST /
2018-10-15 20:01:50,062 - aioftp.client - INFO - 10004:20088 - client.py:169 - 125 Data connection already open; Transfer starting.
2018-10-15 20:01:50,063 - aioftp.client - INFO - 10004:20088 - client.py:169 - 226 Transfer complete.
[]
```

**Comment by toonknapen** [comment]
Indeed, I just instrumented ```AsyncListener.__anext__``` (prefixed with ```line:```) and I get following:

```
2018-10-15 20:07:02,246 - aioftp.client - INFO - 14616:3192 - client.py:240 - LIST /
2018-10-15 20:07:02,247 - aioftp.client - INFO - 14616:3192 - client.py:169 - 125 Data connection already open; Transfer starting.
2018-10-15 20:07:02,248 - aioftp.client - DEBUG - 14616:3192 - client.py:665 - line: b'09-14-18  04:40PM       <DIR>          attrack\r\n'
2018-10-15 20:07:02,248 - aioftp.client - DEBUG - 14616:3192 - client.py:665 - line: b'09-16-18  01:07PM       <DIR>          goodsync\r\n'
2018-10-15 20:07:02,248 - aioftp.client - DEBUG - 14616:3192 - client.py:665 - line: b'03-24-18  04:13PM       <DIR>          SRO XML Feed 2017\r\n'
2018-10-15 20:07:02,248 - aioftp.client - DEBUG - 14616:3192 - client.py:665 - line: b'09-15-18  07:11PM       <DIR>          tmp\r\n'
2018-10-15 20:07:02,248 - aioftp.client - DEBUG - 14616:3192 - client.py:665 - line: b'09-14-18  09:25PM       <DIR>          XML Feed 2018\r\n'
2018-10-15 20:07:02,248 - aioftp.client - DEBUG - 14616:3192 - client.py:665 - line: b'08-15-18  08:47PM           3889244489 XML Feed 2018.zip\r\n'
2018-10-15 20:07:02,249 - aioftp.client - DEBUG - 14616:3192 - client.py:665 - line: b''
```

**Comment by toonknapen** [comment]
the cls.parse_line throws a KeyError

**Comment by pohmelie** [comment]
That is what I was thinking of. There is two problems:
* `LIST` command have no standard, it made for humans.
* aioftp parsing code written by @jw4js targeted only for unix-like output.

So, maybe we can add one more fallback option as trying to parse output new way. I will take a look into ftplib from std lib soon.

**Comment by toonknapen** [comment]
The NLST command is supported. However this command does just list all files and directories and does not provide additional info (tested both on windows 10 and ubuntu 16.04 vsftpd).

I added support for the raw_command=NLST. Have a look at https://github.com/toonknapen/aioftp/commit/5805b0e90850b86782f6fefaf5efc64e4cb0ecee. Do you think this is a good way to go? Or do you prefer to do NLST support in another way. Let me know and eventually I can elaborate on it and do a PR if you like?

**Comment by pohmelie** [comment]
Yes, this can be implemented same way as `LIST` right now, but you need to add `NLST` support on server side too. Add some tests. And don't forget to replace f-strings in logging, since we still support python 3.5.3+.

**Comment by jw4js** [comment]
I can write the Windows parsing code as well for `LIST`. Just to confirm it looks just like `dir` output right?

**Comment by pohmelie** [comment]
@jw4js
Yes, it looks like `dir` output, but threre is no guarantee, that someone will create new issue with «hey, can't list files on my X server with aioftp, when Y client works fine!»
Anyway, it will be good if you implement one more fallback.

**Comment by jw4js** [comment]
How do you want the tests to be implemented? Currently, I directly modified `aioftp/server.py` as a temporary measure to test the code. Should I write the code to generate `dir` output, use `wine cmd` to test, or use another method?

**Comment by pohmelie** [comment]
I think it is enough to test your parsing function directly, even without instantiating of client and server. Just pass some pack of lines and check result.

**Comment by VyachAp** [comment]
Hello everyone, seems we've got some problem with `parse_list_line_windows` function.

```python3
with setlocale("C"):
    strptime = datetime.datetime.strptime
    date_time = strptime(date_time_str, "%m/%d/%Y %I:%M %p")
```
In this case date cannot be parsed properly, so we've got an empty result of `client.list()` function.

We've made some changes to the date format to make it works:
```python3
with setlocale("C"):
    strptime = datetime.datetime.strptime
    date_time = strptime(date_time_str, "%m-%d-%y %I:%M%p")
```

Actually we couldn't understand the role of `setlocale` context, at this time we already have a date as string and setlocale seems doing nothing to this. Probably we should add optional argument for date-time format.
